docs/LibraryVsApplicationTestBundles.md

```diff
@@ -0,0 +1,22 @@
+# Library vs Application Test Bundles
+
+* Authors: [Alan Zeino](https://github.com/alanzeino)
+
+## Introduction
+
+Developers writing unit tests for iOS using XCTest typically use Application Test Bundles without much consideration as to _why_ their tests need to be inside an Application Test Bundle, as opposed to a Library Test Bundle. This document describes what you can and can't do with both, and why you might want to use a Library Test Bundle.
+
+Library Test Bundles were once called _Logic_ Test Bundles in Apple's nomenclature. In the context of this document, both Library and Logic are interchangeable.
+
+### Application tests
+
+Unit tests that test parts of an application (such as UIViewControllers, UIWindows, UIViews) should typically be part of an Application test bundle. An Application test bundle requires a Test Host and at test run time, a Simulator too.
```

Comment:
Library tests also need to be run in the context of a simulator to access ios sdk specific API's. The increased functionality that the test host gives is an app context, which provides access to api's around app/viewcontroller lifecycle, functionality requiring entitlements.